File: generated_code/python/solution_4271_4.py
from typing import List

class Solution:
    def findDegrees(self, matrix: List[List[int]]) -> List[int]:
        """
        Calculates the degree of each vertex in an undirected graph.
        
        Algorithm:
        The degree of a vertex i in an adjacency matrix representation of an 
        undirected graph is equal to the sum of elements in its corresponding row 
        (or column), excluding any self-loops (if matrix[i][i] == 1). Assuming 
        no self-loops as per standard simple graph definitions, it is just the row sum.
        
        Time Complexity: O(n^2) where n is the number of vertices, as we visit each cell once.
        Space Complexity: O(n) to store the result array.
        """
        # The degree of vertex i is simply the sum of 1s in row i
        return [sum(row) for row in matrix]

File: generated_code/cpp/solution_4271_4.cpp
#include <vector>

using namespace std;

class Solution {
public:
    /**
     * Problem: Find the degree of each vertex in an undirected graph given its adjacency matrix.
     * Algorithm: The degree of a vertex i in an adjacency matrix is the count of 1s in row i.
     * Time Complexity: O(n^2) - We must visit each element in the n x n matrix.
     * Space Complexity: O(1) - Excluding the output vector, we use constant extra space.
     */
    vector<int> findDegrees(vector<vector<int>>& matrix) {
        int n = matrix.size();
        vector<int> ans(n, 0);

        for (int i = 0; i < n; ++i) {
            int degree = 0;
            for (int j = 0; j < n; ++j) {
                // If matrix[i][j] is 1, there is an edge between vertex i and vertex j
                if (matrix[i][j] == 1) {
                    degree++;
                }
            }
            ans[i] = degree;
        }

        return ans;
    }
};